### PR TITLE
dtmf sending added

### PR DESCRIPTION
**Pull request by xyzbilal**
Added dtmf functions  to IOS and Android,
peerConnectionId getter added to RTCPeerconnection.dart

Fix https://github.com/flutter-webrtc/flutter-webrtc/issues/292

### CODE REVIEW

File: android/src/main/java/com/cloudwebrtc/webrtc/MethodCallHandlerImpl.java [hunk 1]
```diff
@@ -249,6 +249,33 @@ public void onMethodCall(MethodCall call, @NonNull Result notSafeResult) {
...
+          }
+          if (m_audioSender != null) {
+            DtmfSender dtmfSender = m_audioSender.dtmf();
+            dtmfSender.insertDtmf(tone, duration, gap);
+          }
+          result.success("success");
+        } else {
+          Log.d(TAG, "dtmf() peerConnection is null");
+          result
+              .error("dtmf", "getRemoteDescription() peerConnection is null",
```
Comment (cloudwebrtc) [inline android/src/main/java/com/cloudwebrtc/webrtc/MethodCallHandlerImpl.java:274]: `getRemoteDescription()` change to `sendDtmf()`

File: android/src/main/java/com/cloudwebrtc/webrtc/MethodCallHandlerImpl.java [hunk 2]
```diff
@@ -249,6 +249,33 @@ public void onMethodCall(MethodCall call, @NonNull Result notSafeResult) {
...
         break;
       }
+      case "sendDtmf": {
+        String peerConnectionId = call.argument("peerConnectionId");
+        String tone = call.argument("tone");
+        int duration = call.argument("duration");
+        int gap = call.argument("gap");
+        PeerConnection peerConnection = getPeerConnection(peerConnectionId);
+        if (peerConnection != null) {
+          RtpSender m_audioSender = null;
```
Comment (cloudwebrtc) [inline android/src/main/java/com/cloudwebrtc/webrtc/MethodCallHandlerImpl.java:259]: No need the `m_` prefix.

File: ios/Classes/FlutterWebRTCPlugin.m [hunk 3]
```diff
@@ -239,6 +239,40 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult) result
...
+        NSString* tone = argsMap[@"tone"];
+        NSString* durationString = argsMap[@"duration"];
+        NSString* gapString = argsMap[@"gap"];
+        
+        double duration = [durationString doubleValue];
+        double gap = [gapString doubleValue];
+        RTCPeerConnection *peerConnection = self.peerConnections[peerConnectionId];
+        if(peerConnection) {
+   
+             RTCRtpSender* m_audioSender = nil ;
```
Comment (cloudwebrtc) [inline ios/Classes/FlutterWebRTCPlugin.m:254]: No need the m_ prefix.

**Review by cloudwebrtc — COMMENTED** [review]
@xyzbilal Thanks.
There are just a few minor issues, I can merge them after you modify them.

**Comment by xyzbilal** [comment]
I made corrections. can you check it please.

File: lib/src/rtc_peerconnection.dart [hunk 4]
```diff
@@ -74,6 +74,8 @@ class RTCPeerConnection {
 
   RTCIceConnectionState get iceConnectionState => _iceConnectionState;
 
+  String get peerConnectionId => _peerConnectionId;
```
[the remark this answers is not in the document]
Comment (xyzbilal) [reply lib/src/rtc_peerconnection.dart]: Hi, Actually I am not a perfect coder. I made some changes according to my understanding and according to your advices. I f you are not satisfied please guide me to code according to your will. thanks

**Comment by cloudwebrtc** [comment]
@xyzbilal Don't worry, I have fixed all the problems and passed the ios/android and macOS tests.

**Comment by leean912** [comment]
Hi, is dtmf supported on Windows platform?

**Comment by xyzbilal** [comment]
dtmf sending is android and swift only features that use method channels and triggers in native side.  There is no windows support right now.

**Pull request merged**